src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java

```diff
@@ -1,44 +1,280 @@
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+    }
+
+    @Given("^The REST API \"([^\"]*)\" is up and running$")
+    public void the_REST_API_is_up_and_running(String endPoint) {
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.get(endPoint).accept(MediaType.APPLICATION_JSON);
+        try {
+            result = mockMvc.perform(requestBuilder).andReturn();
+            LOGGER.debug("Response code from mocked REST API: " + String.valueOf(result.getResponse().getStatus()));
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
+    }
+
+    @Given("^Subscriptions are setup using REST API \"([^\"]*)\"$")
+    public void subscriptions_are_setup_using_REST_API(String endPoint) {
+        String readFileToString = "";
+        try {
+            readFileToString = FileUtils.readFileToString(new File(SUBSCRIPTION_WITH_JSON_PATH), "UTF-8");
+            readFileToString = stringReplaceText(readFileToString);
+            readSubscriptionNames(readFileToString);
+            postSubscriptions(readFileToString, endPoint);
+            validateSubscriptionsSuccessfullyAdded(endPoint);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        } 
+    }
+    
+    @Then("^Mail subscriptions were triggered$")
+    public void check_mail_subscriptions_were_triggered() throws Throwable {
+        LOGGER.debug("Verifying received emails.");
+        List<SmtpMessage> emails = smtpServer.getReceivedEmails();
+        assert(emails.size() > 0);
+        
+        for(SmtpMessage email : emails) {
+            // assert correct sender.
+            assertEquals(email.getHeaderValue("From"), sender);
+            // assert given test case exist in body.
+            assert(email.getBody().toString().contains("TC5"));
+        }
+    }
+    
+    @Then("^Rest subscriptions were triggered$")
+    public void check_rest_subscriptions_were_triggered() throws Throwable {
+        LOGGER.debug("Verifying REST requests.");
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_AUTH), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_PARAMS), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_AUTH_PARAMS), Format.JSON))));
+    }
+
+    @When("^I send Eiffel events$")
+    public void send_eiffel_events() {
+        LOGGER.debug("About to send Eiffel events.");
+        try {
+            List<String> eventNames = getEventNamesToSend();
```

Comment:
Use eventNames List to compare with the actual process events to check if all Events has been processed.
If triggers test fail, present the comparison between expected eventNames list and the actual events that has been procssed. Compare also expected aggregated object with the actual aggregated object. With this data we can easier see why triggering failed.